c10/cuda/CMakeLists.txt

```diff
@@ -48,6 +48,8 @@ torch_cuda_based_add_library(c10_cuda ${C10_CUDA_SRCS} ${C10_CUDA_HEADERS})
 set(CUDA_LINK_LIBRARIES_KEYWORD)
 # If building shared library, set dllimport/dllexport proper.
 target_compile_options(c10_cuda PRIVATE "-DC10_CUDA_BUILD_MAIN_LIB")
+# Always enable device assertions.
```

Comment:
Ideally we want to expose it as an option in CMake here: https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt

 but I'm ok landing it like this too

Reply:
Yeah, we can easily expose it as an option later if the need arises.